test/jointjs/graph.js

```diff
@@ -114,640 +114,705 @@ module('graph', {
+        assert.equal(after.rect1.bbox.y, before.rect1.bbox.y + dy, 'position/y of element');
+        assert.equal(after.link1.source.x, before.link1.source.x + dx, 'source/x of link');
+        assert.equal(after.link1.source.y, before.link1.source.y + dy, 'source/y of link');
+        assert.equal(after.link1.target.x, before.link1.target.x + dx, 'target/x of link');
+        assert.equal(after.link1.target.y, before.link1.target.y + dy, 'target/y of link');
+
+        _.each(before.link2.vertices, function(vertexBefore, index) {
+            var vertexAfter = after.link2.vertices[index];
+            assert.equal(vertexAfter.x, vertexBefore.x + dx, 'vertex/x of link');
+            assert.equal(vertexAfter.y, vertexBefore.y + dy, 'vertex/y of link');
+        });
+
+        assert.equal(after.embeddedElement1.bbox.x, before.embeddedElement1.bbox.x + dx, 'position/x of embedded element');
+        assert.equal(after.embeddedElement1.bbox.y, before.embeddedElement1.bbox.y + dy, 'position/y of embedded element');
+        assert.equal(after.embeddedLink1.source.x, before.embeddedLink1.source.x + dx, 'source/x of embedded link');
+        assert.equal(after.embeddedLink1.source.y, before.embeddedLink1.source.y + dy, 'source/y of embedded link');
+        assert.equal(after.embeddedLink1.target.x, before.embeddedLink1.target.x + dx, 'target/x of embedded link');
+        assert.equal(after.embeddedLink1.target.y, before.embeddedLink1.target.y + dy, 'target/y of embedded link');
     });
 
-    rect2.embed(embeddedLink1);
-
-    this.graph.addCells([rect1, rect2, link1, link2, embeddedElement1, embeddedLink1]);
-
-    var dx;
-    var dy;
-    var before;
-    var after;
-
-    dx = 10;
-    dy = 20;
-
-    before = {
-        rect1: {
-            bbox: rect1.getBBox()
-        },
-        link1: {
-            source: link1.get('source'),
-            target: link1.get('target')
-        },
-        link2: {
-            vertices: link2.get('vertices')
-        },
-        embeddedElement1: {
-            bbox: embeddedElement1.getBBox()
-        },
-        embeddedLink1: {
-            source: embeddedLink1.get('source'),
-            target: embeddedLink1.get('target')
-        }
-    };
-
-    this.graph.translate(dx, dy);
-
-    after = {
-        rect1: {
-            bbox: rect1.getBBox()
-        },
-        link1: {
-            source: link1.get('source'),
-            target: link1.get('target')
-        },
-        link2: {
-            vertices: link2.get('vertices')
-        },
-        embeddedElement1: {
-            bbox: embeddedElement1.getBBox()
-        },
-        embeddedLink1: {
-            source: embeddedLink1.get('source'),
-            target: embeddedLink1.get('target')
-        }
-    };
-
-    assert.equal(after.rect1.bbox.x, before.rect1.bbox.x + dx, 'position/x of element');
-    assert.equal(after.rect1.bbox.y, before.rect1.bbox.y + dy, 'position/y of element');
-    assert.equal(after.l
```

Comment:
For consistency, I think this should be a separate method called `graph.resize()` which resizes all the cells in the graph. `graph.resizeCells` would behave similarly to `graph.addCells`.